mapping_parenting_tech/analysis/prototyping/play_store/get_play_store_apps.py

```diff
@@ -160,63 +227,86 @@ def app_snowball(seed_app_id: str, depth: int = 5, __current_depth: int = 1) ->
+    logging.info(f"Attempting to insert {len(app_id_list)} to app list")
 
-    with open(output_target, "wt", newline="\n") as output_file:
-        csv_writer = csv.writer(output_file)
-        csv_writer.writerow([app_category])
-        # NB can't use writerows() as it will treat each string as a list and split it into comma-separated letters
-        # hence why `app_id` (and `app_category`, above) is encapsulated in square brackets
-        for app_id in app_list:
-            csv_writer.writerow([app_id])
+    app_id_df = pd.read_csv(DATA_DIR / "all_app_ids.csv", index_col=None, header=0)
+    app_ids = app_id_df[app_id_df.columns[0]].to_list()
 
-    logging.info(f"App ids saved in {output_target}")
+    orig_length = len(app_ids)
+    app_ids.extend(app_id_list)
 
-    # if we got here, it's all ok so return True
-    return True
+    logging.info(f"Added {len(set(app_ids)) - orig_length} new app ids")
+
+    if dry_run == False:
+        app_id_df = pd.DataFrame(set(app_ids), columns=["appId"])
+        app_id_df.to_csv(DATA_DIR / "all_app_ids.csv", index=False)
+        logging.info(f"App id list saved successfully")
+
+    return set(app_ids)
 
 
 # %%
-def load_app_ids(filename: str, file_path: Path = APP_IDS_PATH) -> tuple:
+def load_all_app_ids() -> set:
     """
-    Loads a pickled list of app ids from a given file and returns the list object
+    Loads the saved ids of all apps and returns them in a set:
 
-    Args:
-        filename: str - file name of pickle file to load within PARENT_DIR
-        file_path: Path - location of the file given in `filename`; default is `outputs/data`
+    Takes no arguments; returns a single set.
+    """
+
+    app_ids_df = pd.read_csv(DATA_DIR / "all_app_ids.csv", index_col=None, header=0)
+    return set(app_ids_df[app_ids_df.columns[0]].to_list())
+
+
+# %%
+def is_app_in_list(app_list: list) -> list:
+    """
+    Identifies app ids that have been saved.
+
+    Arguments
+        app_list: list - a list of app ids to be checked
 
     Returns:
-        str: the name of the app category
-        list: list of app ids
+        list of boolean values, reflecting whether the app id in the equivalent position in `app_list` is present.
+
+    For example:
+        If we want to check three apps, we pass their ids to the function:
+        check_apps = is_app_in_list(["app 1 id", "app 2 id", app 3 id"])
 
+        If only the id for app 3 is in the list, `check_apps` will return:
+        [False, False, True]
     """
 
-    app_id_list = []
-    with open(file_path / filename, "rt") as id_list:
-        csv_reader = csv.reader(id_list)
-        for app_id in csv_reader:
-            app_id_list.extend(app_id)
+    all_app_ids = load_all_app_ids()
+    r_list = list()
 
-    app_category = app_id_list.pop(0)
-    return (app_category, app_id_list)
+    for x in app_list:
+        to_add = True if x in all_app_ids else False
+        r_list.append(to_add)
+
+    return r_list
```

Comment:
You could simplify these lines as: 
```python
return [app_id in all_app_ids for app_id in app_list]
```
(but perhaps test this before you accept the suggestion, in case something doesn't break down further down the line)

Reply:
Think that this is now obsolete...